diff -r d941a4b0a29f configure
--- a/configure	Tue Jun 04 02:57:31 2013 +0800
+++ b/configure	Tue Jun 04 04:11:24 2013 +0800
@@ -25,16 +25,20 @@
 
 stdlib=`ocamlc -where`
 
+LIBDIR=/app/vendor/gdbm/lib
+
 hasgot() {
+  echo "testing hasgot $1 $2 $3 $4"
   rm -f hasgot.c
   (echo "#include <$2>"
    echo 'int main() {'
    echo '  (void) dbm_open("foo", 0, 0);'
    echo '  return 0;'
    echo '}') > hasgot.c
-  ${CC:-cc} -I$1 -o hasgot.exe hasgot.c $3 $4
+  ${CC:-cc} -L$LIBDIR -I$1 -o hasgot.exe hasgot.c $3 $4
   res=$?
   rm -f hasgot.c hasgot.exe
+  echo "tested hasgot $1 $2 $3 $4: $res"
   return $res
 }
 
@@ -42,19 +46,19 @@
 dbm_link="not found"
 dbm_defines=""
 
-for dir in /usr/include /usr/include/db1 /usr/include/gdbm; do
+for dir in /usr/include /usr/include/db1 /usr/include/gdbm /app/vendor/gdbm/include; do
   if test -f $dir/ndbm.h; then
     dbm_include=$dir
     if hasgot $dir ndbm.h; then
       dbm_link=""
     elif hasgot $dir ndbm.h -lndbm; then
-      dbm_link="-lndbm"
+      dbm_link="-L$LIBDIR -lndbm"
     elif hasgot $dir ndbm.h -ldb1; then
-      dbm_link="-ldb1"
+      dbm_link="-L$LIBDIR -ldb1"
     elif hasgot $dir ndbm.h -lgdbm; then
-      dbm_link="-lgdbm"
+      dbm_link="-L$LIBDIR -lgdbm"
     elif hasgot $dir ndbm.h -lgdbm_compat -lgdbm; then
-      dbm_link="-lgdbm_compat -lgdbm"
+      dbm_link="-L$LIBDIR -lgdbm_compat -lgdbm"
     fi
     break
   fi
@@ -62,7 +66,7 @@
     dbm_include=$dir
     dbm_defines="-DDBM_USES_GDBM_NDBM"
     if hasgot $dir gdbm-ndbm.h -lgdbm_compat -lgdbm; then
-      dbm_link="-lgdbm_compat -lgdbm"
+      dbm_link="-L$LIBDIR -lgdbm_compat -lgdbm"
     fi
     break
   fi
